File: 1189/solution.java
class Solution {
    public int maxNumberOfBalloons(String text) {
        Map<Character, Integer> map = new HashMap<>();
        map.put('b',0);
        map.put('a',0);
        map.put('l',0);
        map.put('o',0);
        map.put('n',0);
        char c;
        for(int i=0;i<text.length();i++){
            c = text.charAt(i);
            if(map.containsKey(c))
                map.put(c, map.get(c)+1);
        }
        int res = map.get('a');
        map.put('l', map.get('l')/2);
        map.put('o', map.get('o')/2);
        for(char ch: map.keySet())
            res = Math.min(res, map.get(ch));
        
        return res;
    }
}

File: 1189/solution.py
class Solution:
    def maxNumberOfBalloons(self, text: str) -> int:
        cnt = {'b':0, 'a':0, 'l':0,'o':0, 'n':0 }
        for i in text:
            if i in cnt:
                cnt[i]+=1
        res = min(cnt['b'], cnt['a'], cnt['l']//2, cnt['o']//2, cnt['n'])
        return res
